pkg/sdk/identifier_parsers.go

```diff
@@ -37,6 +38,10 @@ func ParseIdentifierString(identifier string) ([]string, error) {
 		if strings.Contains(part, `"`) {
 			return nil, fmt.Errorf(`unable to parse identifier: %s, currently identifiers containing double quotes are not supported in the provider`, identifier)
 		}
+		// TODO(SNOW-1571674): Remove the validation
+		if strings.ContainsAny(part, `()`) {
+			return nil, fmt.Errorf(`unable to parse identifier: %s, currently identifiers containing opening and closing parentheses '()' are not supported in the provider`, identifier)
+		}
```

Comment:
Do we have documented such behaviors? I think it would be nice to keep them in one place that users can easily access.

Reply:
True, I'll create a ticket for that to sum up the identifiers rework in our changes before v1 GH document.

Reply:
Update: created SNOW-1616523